pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -148,9 +178,23 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 	}
 
 	nextIndex := (b.firstIndex + b.size) % len(b.points)
-	b.points[nextIndex] = p
 	b.size++
-	return nil
+	return &b.points[nextIndex], nil
+}
+
+// Remove the last point that was allocated.
+// This is used for when NextPoint allocates a point that is then unused and
+// needs to be returned to the ring buffer.
+// This occurs when a histogram point has a stale marker.
```

Comment:
Might be worth adding a note that the method will panic if there are no points in the buffer.